build-system/tasks/prettify.js

```diff
@@ -46,7 +47,10 @@ const tempDir = tempy.directory();
  * Checks files for formatting (and optionally fixes them) with Prettier.
  */
 async function prettify() {
-  const filesToCheck = getFilesToCheck(prettifyGlobs, {dot: true});
+  const ignored = ignore().add(fs.readFileSync('.prettierignore', 'utf8'));
+  const filesToCheck = ignored.filter(
+    getFilesToCheck(prettifyGlobs, {dot: true})
```

Comment:
`getFilesToCheck` calls [`globby.sync`](https://github.com/sindresorhus/globby#options), which uses [`fast-glob`](https://github.com/mrmlnc/fast-glob#ignore) under the covers. It should be sufficient to add an `ignore` field to the options object and pass in the list of lines in `.prettierignore`.

An alternative would be to add a new `ignoreFile` param to `getFilesToCheck` and have it read the file and construct the list of globs to ignore and pass them in to `globby.sync({ignore: ...})`.

Reply:
I didn't want to add ignore file parsing to our code (comments, rules, etc). `ignore` is what both [prettier](https://github.com/prettier/prettier/blob/a17d3fb21f3dc45774f41a313d384ea156ae5b92/package.json#L54) and [eslint](https://github.com/eslint/eslint/blob/8984c91372e64d1e8dd2ce21b87b80977d57bff9/package.json#L67) use internally to create their ignore files, so it's what I went with.

Reply:
In that case, WDYT about the alternative suggested above? (Add an optional param called `ignoreFile` to `getFilesToCheck()` and if present, ignore the files listed in it.)

The goal is to be able to reuse this logic in other tasks that have their own ignore files. (e.g. lint can use this with `.eslintignore`)

Reply:
Yah, I can relocate, but it's only prettier that needs it. (Prettier ignores the `.prettierignore` when using the API, and only uses it when using the CLI). Eslint and friends do the right thing. I'll add a comment explaining.